frontend/src/components/transaction-manager/transaction-manager.tsx

```diff
@@ -0,0 +1,128 @@
+import * as Sentry from '@sentry/react'
+import * as React from 'react'
+
+import { NEW_CONSENT_REQUEST } from '../../lib/events'
+import { Service } from '../../service'
+import type { ConsentRequest } from '../../wailsjs/go/models'
+import { TransactionModal } from '../transaction-modal'
+import type {
+  ParsedTx,
+  Transaction,
+  TransactionPayload
+} from './transaction-types'
+
+/**
+ * Stores an array of parsed transactions which get passed to a modal
+ */
+export function TransactionManager() {
+  const [transactions, setTransactions] = React.useState<ParsedTx[]>([])
+
+  const handleResponse = React.useCallback(
+    async (txId: string, decision: boolean) => {
+      try {
+        await Service.ConsentToTransaction({
+          txId,
+          decision
+        })
+
+        // Remove the rejected/approved transaction
+        setTransactions(curr => curr.filter(tx => tx.txId !== txId))
+      } catch (err) {
+        Sentry.captureException(err)
+      }
+    },
+    []
+  )
+
+  // Get any already pending tx on startup
+  React.useEffect(() => {
+    const run = async () => {
+      try {
+        const res = await Service.ListConsentRequests()
+        setTransactions(res.requests.map(parseTx))
+      } catch (err) {
+        Sentry.captureException(err)
+      }
+    }
+
+    // Listen for new incoming transactions
+    window.runtime.EventsOn(NEW_CONSENT_REQUEST, (tx: ConsentRequest) => {
+      setTransactions(curr => [...curr, parseTx(tx)])
+    })
+
+    run()
+  }, [])
+
+  const orderedTransactions = React.useMemo(() => {
+    if (!transactions.length) {
+      return []
+    }
+
+    // Oldest at index 0
+    return transactions.sort((a, b) => {
+      return a.receivedAt.getTime() - b.receivedAt.getTime()
```

Comment:
nitpick (non-blocking): not sure if recivedAt will be continuous or discrete (i.e. is it a block time or a general time), if continuous secondary sort on id would be a good idea

Reply:
It's the date on when the consent request has been received by the wallet app. It's just something to sort the transactions.

Sorting on something else might not be practical from the user POV.